subsys/ipc/ipc_service/backends/ipc_rpmsg.c

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2021 Carlo Caione <ccaione@baylibre.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <ipc/ipc_rpmsg.h>
+
+static void rpmsg_service_unbind(struct rpmsg_endpoint *ep)
+{
+	rpmsg_destroy_ept(ep);
+}
+
+static void ns_bind_cb(struct rpmsg_device *rdev, const char *name, uint32_t dest)
+{
+	struct rpmsg_virtio_device *p_rvdev;
+	struct ipc_rpmsg_instance *instance;
+	struct ipc_ept *ept;
+	int err;
+
+	p_rvdev = CONTAINER_OF(rdev, struct rpmsg_virtio_device, rdev);
+	instance = CONTAINER_OF(p_rvdev->shpool, struct ipc_rpmsg_instance, shm_pool);
+
+	for (size_t i = 0; i < NUM_ENDPOINTS; i++) {
+		ept = &instance->endpoint[i];
+
+		if (strcmp(name, ept->name) == 0) {
+			err = rpmsg_create_ept(&ept->ep, rdev, name, RPMSG_ADDR_ANY,
+					       dest, instance->cb, rpmsg_service_unbind);
+			if (err != 0) {
+				return;
+			}
+
+			ept->bound = true;
+			if (instance->bound_cb) {
+				instance->bound_cb(ept);
+			}
+		}
+	}
+}
+
+int ipc_rpmsg_register_ept(struct ipc_rpmsg_instance *instance, struct ipc_ept *ept)
+{
+	int err;
+
+	if (!instance || !ept) {
+		return -EINVAL;
+	}
+
+	if (!IS_ENABLED(CONFIG_IPC_SERVICE_MASTER)) {
+		err = rpmsg_create_ept(&ept->ep, instance->rdev, ept->name, RPMSG_ADDR_ANY,
+				       RPMSG_ADDR_ANY, instance->cb, rpmsg_service_unbind);
+		if (err != 0) {
+			return err;
+		}
+	}
+
+	return 0;
+}
+
+int ipc_rpmsg_init(struct ipc_rpmsg_instance *instance, void *shb, size_t size)
+{
+	int err;
+
+	if (!instance || !shb) {
+		return -EINVAL;
+	}
+
+	if (IS_ENABLED(CONFIG_IPC_SERVICE_MASTER)) {
```

Comment:
( commented in my previous review but seems not to have been integrated in my review)

How do you address following use case with this static CONFIG_IPC_SERVICE_MASTER config?
core 1 (master) <-- IPC1 --> core 2(remote)
core 2 (master) <-- IPC2 --> core 3(remote)
In such case the core 2 should implement the IPC service with a  multicore backend with 2 different roles.
A role parameter should be added to ipc_rpmsg_init function to be more dynamic ( can be added in your PR introducing multi-core backend)

Reply:
Yes, this is a good idea. And the role should also be specified in the DT representation of the instance.